woocommerce/woocommerce-android
# Analytics Hub:  Integrate Analytics Hub with new range implementation

Changelog
==========
Adjusts the `AnalyticsViewModel` and `AnalyticsRepository` to rely on the Selection class implementation as the source of truth for date range information while refactoring some internal operations in both classes that no longer make sense after the adoption of this new implementation.

It also: 
- Adjusts and simplifies both classes' unit test implementations while keeping all scenarios and assertions.
- Replace usage of legacy types around the app when calling the Analytics Hub for presentation with the new ones introduced by the Selection class.

⚠️ The Date range selector UI is not displaying all information correctly after the changes introduced here. I decided to leave this in a dedicated PR alongside the UI adjustment of the same range selector here: https://github.com/woocommerce/woocommerce-android/pull/8122 

Disclaimer
==========
Since it will take a bunch of changes to achieve the complete scope, this PR is targeted to a feature branch called `improve/analytics-hub` that will aggregate everything, this way we can review piece by piece while keeping the production version intact.

How to Test
==========
1. Build the app and open the Analytics Hub. If possible, keep a second device/emulator running the Hub from the `trunk` branch
2. Verify that Hub still opens following the selected tab from the My Store view. E. g. `This Year` tab should open the Hub selected in the `Year to Date` range.
3. Verify that all options for date range still exist and are the same from the original Hub
4. Verify that all ranges display the same data from the original Hub inside the stats cards

Update release notes:
